### PR TITLE
Update tests to work with nightly latest build

### PR DESCRIPTION
**Pull request by mkleen**
## Summary of the changes / Why this is an improvement

This adapts the tests to the following changes:
- https://github.com/crate/crate/commit/608c1e8bd68678f35bf5f980398d3cc0a74e4977
- https://github.com/crate/crate/commit/468a392eea3b86cb3ac5322abbc4515b2269027e
- https://github.com/crate/crate/pull/9312/commits/8451a00e631d5dcc5c79df09f064c0171501f922
- crate/crate@9d57c83

## Checklist

 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -8,6 +8,7 @@ Unreleased
 Breaking changes
 ================
 
+- Removed outdated metrices from query stats, threadpools and circut breakers.
```
Comment (seut) [inline CHANGES.rst:11]:
```suggestion
- Removed outdated metrics from query stats, threadpools and circuit breakers.

```

File: src/test/java/io/crate/jmx/integrationtests/MetricsITest.java [hunk 2]
```diff
@@ -113,7 +101,7 @@ public void testThreadPoolsMetrics() {
     @Test
     public void testCircuitBreakersMetrics() {
         List<String> names = Arrays.asList(
-                "parent", "query", "jobs_log", "operations_log", "fielddata", "request", "in_flight_requests");
+                "parent", "query", "jobs_log", "operations_log", "request", "in_flight_requests");
```
Comment (seut) [inline src/test/java/io/crate/jmx/integrationtests/MetricsITest.java:104]: Aren't the fielddata breaker stats still exposed?

**Review by seut — COMMENTED** [review]
Looks good in general, added some comments.

As you mentioned it, how is the https://github.com/crate/crate/commit/3b8996d1739e84a9355dbc253389e1f4e184018d related here?

File: src/test/java/io/crate/jmx/integrationtests/MetricsITest.java (hunk 2 above)
Comment (mkleen) [reply src/test/java/io/crate/jmx/integrationtests/MetricsITest.java:104]: They are deprecated and return the default value -1:
https://github.com/crate/crate/commit/9d57c83a5624587d422539f6d2a0267958621926

Comment (mkleen) [reply src/test/java/io/crate/jmx/integrationtests/MetricsITest.java:104]: Should they stay here for now ?

Comment (mfussenegger) [reply src/test/java/io/crate/jmx/integrationtests/MetricsITest.java:104]: I think we should try to keep newer jmx_exporters working with all versions in the current major release series. So every `4.x` version should work with any recent jmx_exporter release.

**Comment by mkleen** [comment]
> Looks good in general, added some comments.
> 
> As you mentioned it, how is the [crate/crate@3b8996d](https://github.com/crate/crate/commit/3b8996d1739e84a9355dbc253389e1f4e184018d) related here?

The jvm metrics in the integration test changed to jdk 15 e.g. switch to G1 garbage collection.
https://github.com/crate/jmx_exporter/blob/139b483d1e702c85c8f568da0f4ffd911d4ee13a/src/test/java/io/crate/jmx/integrationtests/MetricsITest.java#L177

**Comment by mkleen** [comment]
> Looks good in general, added some comments.
> 
> As you mentioned it, how is the [crate/crate@3b8996d](https://github.com/crate/crate/commit/3b8996d1739e84a9355dbc253389e1f4e184018d) related here?

Sorry, my confusion here. I guess this commit https://github.com/crate/crate/pull/9312/commits/8451a00e631d5dcc5c79df09f064c0171501f922 to the change in the jmx jvm metrics.

File: src/test/java/io/crate/jmx/integrationtests/MetricsITest.java (hunk 2 above)
Comment (mkleen) [reply src/test/java/io/crate/jmx/integrationtests/MetricsITest.java]: Ok, thanks for the feedback. I adapted the `CrateCollectorTest` that it works with all 4.x CrateDB versions, and the `MetricsITest` that it works with `latest-nightly`. All deprecated outdated metrics are added back. So this change only affects only the tests.

File: src/test/java/io/crate/jmx/integrationtests/MetricsITest.java [hunk 3]
```diff
@@ -129,8 +117,8 @@ private void assertMetricValue(String metricString) {
         int endIdx = METRICS_RESPONSE.indexOf("\n", startIdx);
         String metricValueStr = METRICS_RESPONSE.substring(startIdx + metricString.length(), endIdx);
         assertThat(
-                metricValueStr + " must be in the format \\d+.\\d+(E\\d+)?",
-                metricValueStr.matches("\\d+.\\d+(E\\d+)?"),
+                String.format("Metric '%s' value '%s' must be in the format ^-\\d+.\\d+(E\\d+)?", metricString, metricValueStr),
+                metricValueStr.matches("^-?\\d+.\\d+(E\\d+)?"),
```
Comment (mkleen) [inline src/test/java/io/crate/jmx/integrationtests/MetricsITest.java:121]: negative values are allowed now for deprecated metrics such as `fielddata`.

**Comment by mkleen** [comment]
Commits will be squashed together once it is approved.

**Review by seut — APPROVED** [review]
lgtm

**Review by mfussenegger — APPROVED** [review]

**Pull request merged**